File: 128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> hs = new HashSet<>();
        for(int num : nums) hs.add(num);
        int ans = 0;

        for(int i = 0; i < nums.length; i++){
            int num = nums[i];
            if(!hs.contains(num)) continue;

            int left = num - 1;
            int countLeft = 0;
            while(hs.contains(left)){
                hs.remove(left);
                left--;
                countLeft++;
            }

            int right = num + 1;
            int countRight = 0;
            while(hs.contains(right)){
                hs.remove(right);
                right++;
                countRight++;
            }

            ans = Math.max(ans, countLeft + 1 + countRight);
            hs.remove(num);
        }

        return ans;
    }
}

File: 128. Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> hs;
        for (int num : nums) hs.insert(num);

        int ans = 0;
        for (int num : nums) {
            if (!hs.count(num)) continue;

            int left = num - 1, right = num + 1;

            while (hs.count(left)) {
                hs.erase(left);
                left--;
            }

            while (hs.count(right)) {
                hs.erase(right);
                right++;
            }

            hs.erase(num);
            ans = max(ans, 1 + (num - left - 1) + (right - num - 1));
        }

        return ans;
    }
};